lib/Serialization/ModuleFile.h

```diff
@@ -860,14 +865,30 @@ class ModuleFile
   /// Returns the generic signature for the given ID.
   GenericSignature *getGenericSignature(serialization::GenericSignatureID ID);
 
+  /// Returns the generic signature for the given ID or the first error.
+  llvm::Expected<GenericSignature *> getGenericSignatureChecked(
+      serialization::GenericSignatureID ID);
+
   /// Returns the substitution map for the given ID, deserializing it if
   /// needed.
   SubstitutionMap getSubstitutionMap(serialization::SubstitutionMapID id);
 
+  /// Returns the substitution map for the given ID, deserializing it if
+  /// needed, or the first error.
+  llvm::Expected<SubstitutionMap> getSubstitutionMapChecked(
+                                     serialization::SubstitutionMapID id);
+
   /// Recursively reads a protocol conformance from the given cursor.
   ProtocolConformanceRef readConformance(llvm::BitstreamCursor &Cursor,
                                          GenericEnvironment *genericEnv =
                                            nullptr);
+
+  /// Recursively reads a protocol conformance from the given cursor,
+  /// returns the conformance or the first error.
+  llvm::Expected<ProtocolConformanceRef> readConformanceChecked(
+                                         llvm::BitstreamCursor &Cursor,
+                                         GenericEnvironment *genericEnv =
```

Comment:
Nitpick: I know you're matching indentation that's already there, but LLVM style would break after the return type so you can do normal align-to-open-paren indentation. ("whatever `clang-format` does" is ultimately our gold standard here.)